src/main/java/sg/ncl/testbed_interface/Dataset.java

```diff
@@ -14,28 +19,40 @@
 public class Dataset implements Serializable {
 
 	private Integer id;
+    @NotEmpty
 	private String name;
+    @NotEmpty
 	private String description;
 	private String contributorId;
-	private String visibility;
-	private String accessibility;
-	private String releaseDate;
+	private DataVisibility visibility;
+	private DataAccessibility accessibility;
+    private ZonedDateTime releasedDate;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Make this value-based field transient so it is not included in the serialization of this class. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3437)